src/irr/video/COpenGLDescriptorSet.h

```diff
@@ -0,0 +1,166 @@
+#ifndef __IRR_C_OPENGL_DESCRIPTOR_SET_H_INCLUDED__
+#define __IRR_C_OPENGL_DESCRIPTOR_SET_H_INCLUDED__
+
+#include "irr/video/IGPUDescriptorSet.h"
+#include "irr/macros.h"
+#include "COpenGLExtensionHandler.h"
+#include "COpenGLBuffer.h"
+#include "COpenGL1DTexture.h"
+#include "COpenGL1DTextureArray.h"
+#include "COpenGL2DTexture.h"
+#include "COpenGL2DTextureArray.h"
+#include "COpenGLCubemapTexture.h"
+#include "COpenGLCubemapArrayTexture.h"
+#include "COpenGL3DTexture.h"
+#include "COpenGLBufferView.h"
+
+namespace irr {
+namespace video
+{
+
+class COpenGLDescriptorSet : public IGPUDescriptorSet
+{
+public:
+    struct SMultibindParams
+    {
+        struct SMultibindBuffers
+        {
+            const GLuint* buffers;
+            const GLintptr* offsets;
+            const GLsizeiptr* sizes;
+        };
+        struct SMultibindTextures
+        {
+            const GLuint* textures;
+            const GLenum* targets;//for when ARB_multi_bind isn't there
+        };
+        struct STexImageBindings //in this case ARB_multi_bind is useless
```

Comment:
its not, layer and levels can be 0 if you use a texture view. `layered` is always true for ARB_multibind and the shader itself declares the access and format.

Reply:
can use same struct as `SMultibindTextures`

Reply:
resolved